src/sidebar/components/annotation-quote.js

```diff
@@ -22,6 +24,9 @@ function AnnotationQuote({ annotation, settings = {} }) {
   // as part of the annotation anchoring process.
   const documentLanguage = '';
 
+  // IE11 does not support `dir=` attribute.
+  const dirProp = useMemo(() => (isIE11() ? {} : { dir: 'auto' }), []);
```

Comment:
https://www.w3.org/International/tests/repo/results/the-dir-attribute-auto

 IE11 is not the easiest to debug or even understand but when adding the dir= attribute to an `<input>`, for example, it no longer renders in a predictable way.  The layout will be severely broken and more or less unusable (or at least the element with that attribute is unusable). I see no way of getting IE11 to work when adding this attribute to an input or textarea so I think its best left to a conditional while we support the browser.

Reply:
> IE11 is not the easiest to debug or even understand but when adding the dir= attribute to an <input>, for example, it no longer renders in a predictable way. 

I found https://stackoverflow.com/questions/38638718 this morning and was able to reproduce. Attempting to do `someInputElement.dir = 'auto'` causes an exception in IE. You can set it to "ltr", "rtl" or an empty string though.

I'm guessing what you saw was the result of this exception in the middle of an update causing all further UI processing to stop and leaving the UI in an arbitrarily broken state, which admittedly is super confusing if you haven't seen that there is an error in the browser console.

On this basis though, you're right that we need to not set `dir` to `auto` in IE. It looks to me like setting it to an empty string should behave the same way as not setting it. In that case you could do something like:

```js
// `dirAuto` is a constant that is defined as the string "auto" in modern browsers
// and the empty string in IE.
import { dirAuto } from '../util/dom'

function SomeWidget() {
  return <input dir={dirAuto} />
}
```

Reply:
I went with Method C, mostly because I was attempted the renderer-option with the key remapping (below) and we can limit it to only taking effect for IE11 so it won't harm performance for anything else.